src/NUnitFramework/framework/Internal/Execution/TestWorker.cs

```diff
@@ -157,7 +158,13 @@ public void Start()
             _workerThread = new Thread(new ThreadStart(TestWorkerThreadProc));
             _workerThread.Name = Name;
 #if APARTMENT_STATE
-            _workerThread.SetApartmentState(WorkQueue.TargetApartment);
+            try
+            {
+                _workerThread.SetApartmentState(WorkQueue.TargetApartment);
+            }
+            catch (PlatformNotSupportedException)
+            {
+            }
```

Comment:
I'm not sure silently catching the exception here is ok. I think this will mean that whenever tests are running in parallel, then STA tests will end up running on a thread which is not STA, with no error. 

I'm not sure of the correct solution to this problem, without digging into the code more I'm afraid - this isn't part of the code I'm too familiar

Reply:
There's probably a better way to do this, but this was the smallest change I could think of at the time. It works for both sequential and parallel. What happens is that the work item starts with the wrong apartment state and errors itself out when it finds out. (Should be commented.)